src/Contracts/SettingContract.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ * This file is part of AlgoliaSearch Client PHP Helper.
+ *
+ * (c) Algolia Team <contact@algolia.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace Algolia\AlgoliaSearch\Helper\Contracts;
+
+/**
+ * @internal
+ */
+interface SettingContract
+{
+    /**
+     * Checks if the given key/value is a setting.
+     *
+     * @param string                         $key
+     * @param array<int, string>|string|null $value
+     * @param array<int, string>             $setting
+     *
+     * @return array<int, string>
+     */
+    public function getValue($key, $value, $setting);
```

Comment:
to me this code and its implementations could be improved in two ways:
- the function is not really named after what it does
- in the implementation `$setting` as a whole are returned and reassigned to itself. maybe there could be another way.
wdyt?